### PR TITLE
NVCTS-604 Add WireMock support

### PR DESCRIPTION
**Pull request by lehphyro**
Introduce a new mode called `PERFORMANCE` that spins up [WireMock](http://wiremock.org/) instead of Prism.
WireMock allows us to mock requests/responses through [stub mapping files](http://wiremock.org/docs/stubbing/) and supports [high performance mocking](http://wiremock.org/docs/configuration/#jetty-configuration) that are suitable for use in performance tests. It also supports introduction of [delays in responses](http://wiremock.org/docs/simulating-faults/) to simulate realistic and performance degradation scenarios.

It'll allow us to mock out all dependencies of a service (access-control, arrangement-manager, etc) in performance tests to create consistent results and avoid issues caused by dependencies.

### CODE REVIEW

**Comment by lehphyro** [comment]
I've configured a docker-compose file and stub mappings that allow creation of a pocket to work successfully: https://stash.backbase.com/projects/CTRLSDLC/repos/retail-ams-local-testing/browse?at=refs%2Fheads%2Fapi-simulator-performance-mode

Example config:
```
  access-control:
    image: harbor.backbase.eu/development/api-simulator:${API_SIMULATOR_VERSION}
    environment:
      spring.application.name: access-control
      backbase.api.simulator.mode: PERFORMANCE
      backbase.api.simulator.port: 8080
      server.port: 18080
    volumes:
      - ./performance/access-control:/config
```

File: src/test/java/com/backbase/api/simulator/config/ApiSimulatorConfigurations.java [hunk 1]
```diff
@@ -18,12 +17,13 @@ public static ApiSimulatorConfiguration defaultConfig() {
         config.setSpec(
             "https://repo.backbase.com/native/specs/transaction-manager/transaction-manager-client-api-v2.5.0.yaml");
         config.setSpecAuthorizations(ImmutableMap.of("repo.backbase.com", "authorization"));
+        config.setMode(ServerMode.SIMULATION);
         return config;
     }
 
     public static ApiSimulatorConfiguration proxyMode() throws MalformedURLException {
```
Comment (SjaakBall) [inline src/test/java/com/backbase/api/simulator/config/ApiSimulatorConfigurations.java:24]: It seems that a configuration with mode PERFORMANCE is not being used in the unit tests.

File: README.md [hunk 2]
```diff
@@ -53,7 +61,40 @@ Then you can run the service with:
...
+
+Create stub mappings according to [WireMock's documentation](http://wiremock.org/docs/stubbing/).
+
+Stub mappings must be in a directory called `mappings`.
+
+### Locally
+
+Run it as a Spring Boot application with working directory pointing to the root directory of this project.
+Use `com.backbase.api.simulator.Application` as main class, `local` profile and configure it with
+`config/application-local.yml` file.
```
Comment (SjaakBall) [inline README.md:76]: When trying to get a local Spring boot application running, it did not pick up the stubs from the mappings directory. Maybe a little more detailed description would be appropriate here.

File: src/test/java/com/backbase/api/simulator/config/ApiSimulatorConfigurations.java (hunk 1 above)
Comment (lehphyro) [reply src/test/java/com/backbase/api/simulator/config/ApiSimulatorConfigurations.java:24]: Yeah, there isn't much to unit test in PERFORMANCE mode, it just starts/stops WireMock, that'd be an integration test which doesn't add much IMO.

File: README.md (hunk 2 above)
Comment (lehphyro) [reply README.md:76]: Added a config property to make it pick them up if PERFORMANCE mode is enabled.

**Review by SjaakBall — APPROVED** [review]

**Review by richkum92 — APPROVED** [review]

**Review by ugurkaya88 — APPROVED** [review]

**Pull request merged**